agent/eventhandler/task_handler.go

```diff
@@ -55,14 +56,17 @@ func NewTaskHandler() *TaskHandler {
 	}
 }
 
-// AddTaskEvent queues up a state change for sending using the given client.
-func (handler *TaskHandler) AddTaskEvent(change api.TaskStateChange, client api.ECSClient) {
-	handler.addEvent(newSendableTaskEvent(change), client)
-}
+// AddStateChangeEvent queues up a state change for sending using the given client.
+func (handler *TaskHandler) AddStateChangeEvent(change statechange.StateChangeEvent, client api.ECSClient) {
+	switch change.GetEventType() {
+	case statechange.TaskEvent:
+		se := newSendableTaskEvent(change.(api.TaskStateChange))
```

Comment:
Can you use the multi value format instead for casting the type?
`event, ok := change.(api.TaskStateChange)`. You can also return an error if you can't cast it. The single value return panics if it can't cast it. So, something like this:

```
func (handler *TaskHandler) AddStateChangeEvent(change statechange.StateChangeEvent, client api.ECSClient) error {
 switch change.GetEventType() {
 case statechange.TaskEvent:
  event, ok := change.(api.TaskStateChange)
  if !ok {
   return errors.New("event handler: unable to get task event from state change event")
  }
  handler.addEvent(newSendableTaskEvent(event), client)
  return nil
 case ...
```